force-app/main/default/classes/SOQL.cls

```diff
@@ -288,20 +332,29 @@ public with sharing class SOQL {
      */
     public static List<Account> findAccountsByLeadEmailDomains(Set<Id> leadIds) {
         // Query the Leads
-
+        List<Lead> leadsForAccount = [SELECT Email FROM Lead WHERE Id = :leadIds];
         // Create a set of email domains from the Leads
-
+        Set<String> emailDomains = new Set<String>();
         // Loop through each Lead
-
             // Get the email from the Lead
-
             // Check if the email is not null
-
-                // Add the email domain to the set
-
-        
+            // Add the email domain to the set
+        for(Lead leadToLoop : leadsForAccount){
+            if(leadToLoop.Email != null){
+                String leadEmailDomain = leadToLoop.Email.substringAfter('@');
+                emailDomains.add(leadEmailDomain);
+            } 
+        }
         // Query the Accounts
-        return null; // Replace null with the variable you used to store the result
+        List<Account> accountsWithDomain = new List<Account>();
+        for(String email : emailDomains){
+            for(Account acc : [SELECT Id, Website FROM Account WHERE Website LIKE :('%'+ email +'%')]){
```

Comment:
Generally speaking, when you find yourself writing nested for-loops, it's often an indication that there's a better way to solve your requirement. One way would be to consider how you could convert those lead email domains into account websites. (And then query on that list of websites.)